packages/roosterjs-content-model-core/lib/coreApi/addUndoSnapshot.ts

```diff
@@ -11,19 +16,35 @@ import type { AddUndoSnapshot, Snapshot } from 'roosterjs-content-model-types';
  * when undo/redo to this snapshot
  */
 export const addUndoSnapshot: AddUndoSnapshot = (core, canUndoByBackspace, entityStates) => {
-    const { lifecycle, physicalRoot, undo } = core;
+    const { lifecycle, physicalRoot, logicalRoot, undo } = core;
     let snapshot: Snapshot | null = null;
 
     if (!lifecycle.shadowEditFragment) {
         // Need to create snapshot selection before retrieve innerHTML since HTML can be changed during creating selection when normalize table
         const selection = createSnapshotSelection(core);
         const html = physicalRoot.innerHTML;
 
+        if (!entityStates) {
+            // give plugins the chance to share entity states to include in the snapshot
+            const event = <SnapshotLogicalRootEvent>{
```

Comment:
only trigger this event when logical root is under an entity, or itself is entity. Because in >99.999% case we don't need to trigger this event